cfi_backward.adoc

```diff
@@ -128,103 +152,172 @@ instructions on such machines is not supported.
-difference that the base is implicitly `ssp`, the width is implicitly `XLEN`.
-`sspush` performs a store identically to the existing `STORE` instruction with
-the difference that the base is implicitly `ssp`, the width is implicitly `XLEN`.
+The `sspop` and `sspopchk` instructions perform a load identically to the
+existing `LOAD` instruction with the difference that the base is implicitly
+`ssp`, the width is implicitly `XLEN`. 
 
-The `sspush` and `sspop` require the virtual address in `ssp` to have a shadow stack
-attribute (see Shadow Stack Memory Protection). 
+The `sspush` instruction performs a store identically to the existing `STORE`
+instruction with the difference that the base is implicitly `ssp`, the width
+is implicitly `XLEN`.
 
-If the virtual address in `ssp` is not `XLEN` aligned then the instructions cause a
-load or store/AMO address-misaligned exception.
+The `sspush`, `sspopchk`, and `sspop` require the virtual address in `ssp` to
+have a shadow stack attribute (see <<SSMP>>).
+
+Correct execution of `sspush`, `sspopchk`, and `sspop` requires that `ssp`
+refers to idempotent memory, because the core must be able to handle traps
+detected by these instruction (e.g., a page fault) and re-execute them after
+returning from the trap handler. If the memory reference by the `ssp` is not
+idempotent then the `sspush` instruction causes a store/AMO access fault and the
+`sspop`/`sspopchk` instructions cause a load access fault.
+
+If the virtual address in `ssp` is not `XLEN` aligned then the `sspop`/
+`sspopchk` instructions cause a load address-misaligned exception and the
+`sspush` instruction causes store/AMO address-misaligned exception.
 
 [NOTE]
 ====
 Misaligned accesses to shadow stack are not required and enforcing alignment is
 more secure to detect errors in the program.
+
+The `sspopchk` instruction performs a load followed by a check of the loaded
+data value with the link register source. If the check against the link register
+faults and the instruction is restarted by the trap handler then the instruction
+will perform a load again. If the memory from which the load is performed is
+non-idempotent then the second load may cause unexpected side effects. Shadow
+stack instructions require the memory referenced by `ssp` to be idempotent to
+avoid such concerns. Locating shadow stacks in non-idempotent memory (e.g.,
+non-idempotent device memory) is not an expected usage and requiring memory
+referenced by `ssp` to be idempotent does not pose a significant restriction.
 ====
 
-The operation of the `sspush` instructions is as follows:
+The operation of the `sspush` instruction is as follows:
 
 .`sspush` operation
 [source, text]
 ----
-If (xBCFIE = 1)
-    [ssp] = [ssp] - (XLEN/8)   # decrement ssp by XLEN/8
-   *[ssp] = [src]              # Store src value to address in ssp
+if (xBCFIE = 1)
+    [ssp] = [ssp] - (XLEN/8)   # decrement ssp by XLEN/8 and
```

Comment:
I think you're changing this is the later PR but you want the memop before the SSP update?

Reply:
Yes, its in the later PR.